wallet-new/src/Cardano/Wallet/Kernel/Restore.hs

```diff
@@ -298,6 +299,14 @@ updateRestorationInfo :: Kernel.PassiveWallet
 updateRestorationInfo wallet wId upd =
   modifyMVar_ (wallet ^. walletRestorationTask) (pure . M.adjust upd wId)
 
+-- | Clears the restoration state and stops and threads.
+stopAllRestorations :: Kernel.PassiveWallet -> IO ()
+stopAllRestorations pw = do
+    mp <- takeMVar (pw ^. walletRestorationTask)
+    let vals = M.elems mp
+    mapM_ _wriCancel vals
+    putMVar (pw ^. walletRestorationTask) (M.empty)
```

Comment:
Should probably use `modifyMVar` here. `takeMVar .. putMVar` is a bit of an anti-pattern, because if an exception happens in the middle, the `MVar` will be left empty.